src/Commerce/Generator/PayPal/CLI.php

```diff
@@ -50,24 +45,63 @@ public function generate_orders( array $args = array(), array $assoc_args = arra
 
 		for ( $k = 0; $k < $orders_count; $k ++ ) {
 			$user_id   = 0;
-			$ticket_id = $ticket_ids[ array_rand( $ticket_ids, 1 ) ];
-			$ticket    = $paypal->get_ticket( $post_id, $ticket_id );
+			$ticket_qty = array();
+			foreach ( $ticket_ids as $ticket_id ) {
+				$ticket_qty[ $ticket_id ] = mt_rand( $tickets_min, $tickets_max );
+			}
 
-			$this->backup_ticket_total_sales( $ticket_id );
+			$items_data = array();
+			$items_index = 1;
+			$payment_gross  = 0;
 
-			$ticket_qty = mt_rand( $tickets_min, $tickets_max );
+			/** @var \Tribe__Tickets__Commerce__PayPal__Main $paypal */
+			$paypal = tribe( 'tickets.commerce.paypal' );
 
-			$inventory  = $ticket->inventory();
-			$inventory  = - 1 === $inventory ? 'unlimited' : $inventory;
-			$ticket_qty = 'unlimited' === $inventory ? $ticket_qty : min( $ticket_qty, (int) $inventory );
+			foreach ($ticket_ids as $ticket_id) {
+				$this_ticket_qty = $ticket_qty[ $ticket_id ];
+				$post_id   = (int) get_post_meta( $ticket_id, $paypal->event_key, true );
 
-			WP_CLI::log( "Generating an order for {$ticket_qty} attendees for ticket {$ticket_id}, current ticket inventory is {$inventory}" );
+				if ( empty( $post_id ) ) {
+					WP_CLI::error( "Ticket with ID {$ticket_id} is not related to a post, please edit the ticket and retry." );
+				}
 
-			if ( $ticket_qty === 0 ) {
-				WP_CLI::log( "Not generating attendees for ticket {$ticket_id} as it ran out of inventory" );
-				continue;
+				if ( ! $post = get_post( $post_id ) ) {
+					WP_CLI::error( "Ticket with ID {$ticket_id} is related to a post with ID {$post_id} but that post does not exist, please edit the ticket and retry." );
+				}
+
+				$ticket = $paypal->get_ticket( $post_id, $ticket_id );
+
+				$this->backup_ticket_total_sales( $ticket_id );
+
+				$inventory  = $ticket->inventory();
+				$inventory  = - 1 === $inventory ? 'unlimited' : $inventory;
+				$this_ticket_qty = 'unlimited' === $inventory ? $this_ticket_qty : min( $this_ticket_qty, (int) $inventory );
```

Comment:
Hmm ... why not kill the previous line and do `- 1 === $inventory` for the ternary condition on this line?